tests/measurements/test_expval.py

```diff
@@ -127,7 +127,6 @@ def test_observable_is_composite_measurement_value(
         are correct for a composite measurement value."""
         dev = qml.device("default.qubit")
 
-        @qml.defer_measurements
```

Comment:
I feel like rather than not testing `defer_measurements` with finite shots there should be a new test for `defer_measurements`. This is probably also true for other MPs and it would make sense for such tests to be in `tests/transforms/test_defer_measurements.py`. Out of scope for this PR. I've added a shortcut story.

Reply:
I've added some MPs tests for `defer_measurements`.